auth/realm/base/src/main/java/org/wildfly/security/auth/realm/DistributedSecurityRealm.java

```diff
@@ -183,7 +195,19 @@ private boolean nextIdentity() throws RealmUnavailableException {
             }
             currentIdentity = securityRealms[nextRealm].getRealmIdentity(principal);
             nextRealm++;
-            if (!currentIdentity.exists()) {
+            boolean isIdentityExisting = false;
+            try {
+                isIdentityExisting = currentIdentity.exists();
+            } catch (RealmUnavailableException e) {
+                if (!ignoreUnavailableRealms) {
+                    throw e;
+                }
+                ElytronMessages.log.ldapRealmIsNotAvailable(e);
+                if (failoverCallback != null) {
+                    failoverCallback.accept(nextRealm - 1);
```

Comment:
> nextRealm - 1

In case of `FailoverSecurityRealm`, the exception `e` is used, see https://github.com/wildfly-security/wildfly-elytron/blob/5b15da1f22701697a4fa9418f1435d7a535e20e3/auth/realm/base/src/main/java/org/wildfly/security/auth/realm/FailoverSecurityRealm.java#L70

Reply:
With `FailoverSecurityRealm` there is the advantage of knowing which realm will be the failing one, what I'm trying to do here is to pass an index of the failed realm to properly emit `SecurityRealmUnavailableEvent`.

Reply:
What if we use a `BiConsumer` instead so both the exception and index can be used?

Reply:
Actually, just took a look at the corresponding wildfly-core changes. I think your current approach is ok.